modules/core/src/v2/coins/algo.ts

```diff
@@ -135,28 +198,82 @@ export class Algo extends BaseCoin {
   /**
    * Specifies what key we will need for signing - Algorand needs the backup, bitgo pubs.
    */
-  keyIdsForSigning() {
+  keyIdsForSigning(): number[] {
     return [ 0, 1, 2 ]; 
   }
 
+  /**
+   * Explain/parse transaction
+   * @param params
+   * - txBase64: transaction encoded as base64 string
+   * @returns {{displayOrder: [string,string,string,string,string], id: *, outputs: Array, changeOutputs: Array}}
+   */
+  explainTransaction(params: ExplainTransactionOptions): TransactionExplanation {
+    const { txBase64 } = params;
+
+    let tx;
+    try {
+      const txToHex = Buffer.from(txBase64, 'base64');
+      const decodedTx = Encoding.decode(txToHex);
+
+      // if we are a signed msig tx, the structure actually has the { msig, txn } as the root object
+      // if we are not signed, the decoded tx is the txn - refer to partialSignTxn and MultiSig constructor
+      //   in algosdk for more information
+      const txnForDecoding = decodedTx.txn || decodedTx;
+
+      tx = Multisig.MultiSigTransaction.from_obj_for_encoding(txnForDecoding);
+    } catch (ex) {
+      throw new Error('txBase64 needs to be a valid tx encoded as base64 string');
+    }
+
+    const id = tx.txID();
+    const fee = {
+      fee: tx.fee,
+      feeRate: null,
+      size: null,
```

Comment:
Can we make this optional in `TransactionFee` and omit it? Or is expected somewhere else?

Reply:
I think so